app/Http/Controllers/RoomController.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\Room\CreateRequest;
+use App\Http\Requests\Room\UpdateRequest;
+use App\Models\Room;
+use Illuminate\Support\Facades\DB;
+
+class RoomController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $rooms = Room::all();
+
+        return view('rooms.index', compact('rooms'));
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        return view('rooms.create');
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \App\Http\Requests\StoreRoomRequest  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(CreateRequest $request)
+    {
+        $validated = $request->validated();
+        Room::create($validated);
+
+        return redirect()->back()->with('success', trans('Successfully created'));
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  \App\Models\Room  $room
+     * @return \Illuminate\Http\Response
+     */
+    public function show(Room $room)
+    {
+        return view('rooms.show', compact('room'));
+    }
+
+    /**
+     * Show the form for editing the specified resource.
+     *
+     * @param  \App\Models\Room  $room
+     * @return \Illuminate\Http\Response
+     */
+    public function edit(Room $room)
+    {
+        return view('rooms.edit', compact('room'));
+    }
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param  \App\Http\Requests\UpdateRoomRequest  $request
+     * @param  \App\Models\Room  $room
+     * @return \Illuminate\Http\Response
+     */
+    public function update(UpdateRequest $request, Room $room)
+    {
+        $validated = $request->validated();
+        $room->name = $validated['name'];
+        $room->save();
+
+        return redirect()->back()->with('success', trans('Successfully updated'));
+    }
+
+    /**
+     * Remove the specified resource from storage.
+     *
+     * @param  \App\Models\Room  $room
+     * @return \Illuminate\Http\Response
+     */
+    public function destroy(Room $room)
+    {
+        DB::transaction(function () use ($room) {
+            $seats = $room->seats();
+            $screens = $room->screenings();
+            foreach ($seats->get() as $seat) {
+                $seat->tickets()->delete();
```

Comment:
Phần này không nên dùng for để xóa 
có thể lấy ra danh sách ticket theo seat id
`$seatIds = $room->seats->pluck('id');
Ticket::whereIn('seat_id', $seatIds)->delete()`

Như vậy sẽ tốn ít query hơn và tránh n+1